test/puppetlabs/puppetdb/testutils/services.clj

```diff
@@ -192,22 +104,30 @@
            (f)))
        (catch java.net.BindException e
          (log/errorf e "Error occured when Jetty tried to bind to port %s, attempt #%s" port attempts)
-         (puppetdb-instance config services (dec attempts) f))
-       (finally
-         (let [log-contents (slurp log-file)]
-           (when-not (str/blank? log-contents)
-             (utils/println-err
-               "-------Begin PuppetDB Instance Log--------------------\n"
-               log-contents
-               "\n-------End PuppetDB Instance Log----------------------"))))))))
+         (call-with-puppetdb-instance config services (dec attempts) f))))))
 
 (defmacro with-puppetdb-instance
   "Convenience macro to launch a puppetdb instance"
   [& body]
-  `(puppetdb-instance
+  `(call-with-puppetdb-instance
     (fn []
       ~@body)))
 
+(defn call-with-lone-log-suppressed-puppetdb-instance
+  "Calls the call-with-puppetdb-instance with args after suppressing
+  all log events.  If there's an error or worse, prints the log to
+  *err*.  Should not be nested, nor nested with calls to
+  with-log-suppressed-unless-notable."
+  [& args]
+  (with-log-suppressed-unless-notable notable-pdb-event?
+    (apply call-with-puppetdb-instance args)))
+
+(defmacro with-lone-log-suppressed-puppetdb-instance
```

Comment:
Yeah, I noticed that. But with-log-appenders-fn doesn't follow that.

On Tue, Aug 11, 2015 at 12:06 PM, Rob Browning <notifications@github.com>
wrote:

> In test/puppetlabs/puppetdb/testutils/services.clj
> <https://github.com/puppetlabs/puppetdb/pull/1568#discussion_r36785227>:
>
> >      (fn []
> >        ~@body)))
> >
> > +(defn call-with-lone-log-suppressed-puppetdb-instance
> > +  "Calls the call-with-puppetdb-instance with args after suppressing
> > +  all log events.  If there's an error or worse, prints the log to
> > +  *err*.  Should not be nested, nor nested with calls to
> > +  with-log-suppressed-unless-notable."
> > +  [& args]
> > +  (with-log-suppressed-unless-notable notable-pdb-event?
> > +    (apply call-with-puppetdb-instance args)))
> > +
> > +(defmacro with-lone-log-suppressed-puppetdb-instance
>
> Oh, and in this code, I've used the scheme convention that (with-foo ...)
> is the macro that takes a body, and (call-with-foo ...) is the version that
> accepts a function.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/puppetlabs/puppetdb/pull/1568/files#r36785227>.
>